### PR TITLE
Creating cookiejar while creating client

### PR DESCRIPTION
**Pull request by mcardosos**
Fixes autorest tests with cookies
PTAL
@jhendrixMSFT @marstr

### CODE REVIEW

**Comment by marstr** [comment]
Just to be clear, this fixes #181?

**Comment by mcardosos** [comment]
@marstr yes

**Comment by mcardosos** [comment]
@vladbarosan :D

File: autorest/client.go [hunk 1]
```diff
@@ -202,19 +203,21 @@ func (c Client) Do(r *http.Request) (*http.Response, error) {
 	if err != nil {
 		return nil, NewErrorWithError(err, "autorest/Client", "Do", nil, "Preparing request failed")
 	}
-	resp, err := SendWithSender(c.sender(), r)
+
+	if c.Sender == nil {
+		c.Sender = sender()
+	}
+
+	resp, err := SendWithSender(c, r)
```
Comment (jhendrixMSFT) [inline autorest/client.go]: This should have been c.Sender not c and it the cause of the SO.

File: autorest/client.go [hunk 2]
```diff
@@ -202,19 +203,21 @@ func (c Client) Do(r *http.Request) (*http.Response, error) {
...
+
+	resp, err := SendWithSender(c.Sender, r)
 	Respond(resp,
 		c.ByInspecting())
 	return resp, err
 }
 
 // sender returns the Sender to which to send requests.
-func (c Client) sender() Sender {
-	if c.Sender == nil {
```
Comment (jhendrixMSFT) [inline autorest/client.go]: I'm not convinced that removing this from here was a good choice as I now see this nil check sprinkled in various places.

**Pull request merged**